app/src/main/java/com/tari/android/wallet/ui/activity/QRScannerActivity.kt

```diff
@@ -0,0 +1,120 @@
+package com.tari.android.wallet.ui.activity
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.app.Activity
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.widget.Toast
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import butterknife.BindView
+import butterknife.OnClick
+import com.budiyev.android.codescanner.*
+import com.google.zxing.BarcodeFormat
+import com.tari.android.wallet.R
+
+
+const val REQUEST_CAMERA_PERMISSION = 101
+const val EXTRA_QR_DATA="extra_qr_text"
+
+class QRScannerActivity : BaseActivity() {
+
+    override val contentViewId = R.layout.activity_qr_scanner
+    private lateinit var codeScanner: CodeScanner
+
+
+    @BindView(R.id.scanner_view)
+    lateinit var scannerView: CodeScannerView
+
+    @SuppressLint("ClickableViewAccessibility")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        if (ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.CAMERA
+            ) == PackageManager.PERMISSION_DENIED
+        ) {
+            ActivityCompat.requestPermissions(
+                this,
+                arrayOf(Manifest.permission.CAMERA),
+                REQUEST_CAMERA_PERMISSION
+            )
+        } else {
+            startScanning()
+        }
+
+    }
+
+    @OnClick(R.id.qr_close)
+    fun onCloseQrScannerClick() {
+        finish()
+        overridePendingTransition(0, R.anim.slide_down)
+    }
+
+    private fun startScanning() {
+        codeScanner = CodeScanner(this, scannerView)
+
+        codeScanner.camera = CodeScanner.CAMERA_BACK
+        codeScanner.formats = listOf(BarcodeFormat.QR_CODE)
+
+        codeScanner.autoFocusMode = AutoFocusMode.SAFE
+        codeScanner.scanMode = ScanMode.SINGLE
+        codeScanner.isAutoFocusEnabled = true
+        codeScanner.isFlashEnabled = false
+
+        codeScanner.decodeCallback = DecodeCallback {
+            val intent=Intent()
+            intent.putExtra(EXTRA_QR_DATA,it.text)
+            setResult(Activity.RESULT_OK,intent)
+            finish()
+            overridePendingTransition(0, R.anim.slide_down)
+        }
+        codeScanner.errorCallback = ErrorCallback {
+            runOnUiThread {
+                Toast.makeText(
+                    this, "Failed to initialize Camera",
+                    Toast.LENGTH_LONG
+                ).show()
+            }
+        }
```

Comment:
TODO: Pretty error messages and user feedback rather than a toast as per
https://app.zeplin.io/project/5dbdf1a40bdfd3832e35c9c5/screen/5e26abe8de3d8994256095e8